pandas/core/arrays/base.py

```diff
@@ -430,9 +443,10 @@ def __ne__(self, other: Any) -> ArrayLike:  # type: ignore[override]
 
     def to_numpy(
         self,
-        dtype: Optional[Dtype] = None,
+        dtype: Optional[NpDtype] = None,
```

Comment:
I think this is too restrictive, i.e. not allowing for instance `bool` as a type to be passed. The issue is our definition of NpDtype in pandas._typing is too restrictive and needs to be fixed first see https://github.com/pandas-dev/pandas/pull/40412#discussion_r593737779

Reply:
My thoughts on this is that if you are doing a `to_numpy` operation, you should pass a numpy type.  If we leave it as `Dtype`, then some of the `numpy` calls in `to_numpy()` have typing issues, because `Dtype` is too broad.  Maybe the solution is 3 levels of dtype.  One is what a `numpy` method will accept.  The next broadens it to include python types.  The third includes `ExtensionDtype` .  So we could change `_typing.py` to:
```python
NpDtype = Union[str, np.dtype]
TypDtype = Union[NpDtype, Type[Union[str, float, int, complex, bool, object]]
Dtype = Union["ExtensionDtype", TypDtype]
```
and then `TypDtype` (or whatever name we want to use) is the argument type in `to_numpy`

Reply:
> My thoughts on this is that if you are doing a to_numpy operation, you should pass a numpy type.

our docs have `dtype: str or numpy.dtype, optional` for Index, Series and DataFrame, but in reality we accept anything numpy accepts

```
>>> s = pd.Series([1,2,3], dtype="Int64")
>>> s
0    1
1    2
2    3
dtype: Int64
>>>
>>> s.to_numpy(bool)
array([ True,  True,  True])
>>>
```
although technically we could restrict types in public methods to just Union[str, np.dtype] and in EA.to_numpy to just np.dtype and convert in Index, Series and DataFrame, do we want to do this to appease mypy?

if so, we should mention it in the release notes.

Reply:
> although technically we could restrict types in public methods to just Union[str, np.dtype] and in EA.to_numpy to just np.dtype and convert in Index, Series and DataFrame, do we want to do this to appease mypy?
> 

I could try redefining `NpDtype` and see what happens???  Let me know.  Or create the intermediate `TypDtype` and use that just in the signature of `to_numpy()`

Reply:
> I could try redefining `NpDtype` and see what happens???

yep, we just need to import the dtype alias from numpy to fix things here as a precursor.

Reply:
> > I could try redefining `NpDtype` and see what happens???
> 
> yep, we just need to import the dtype alias from numpy to fix things here as a precursor.

But if we do that import, then we are creating a dependency on numpy >= 1.20.1 .   Is that OK?

Reply:
we may need some import gymnastics in pandas._typing to fall back to either the current alias or just `Any` if numpy < 1.20 is installed.

Reply:
I think that should be saved for another PR.  What do you think?

Reply:
definitely. This PR is still too big. if done as a pre-cursor may reduce the changes here.